src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java

```diff
@@ -162,6 +169,21 @@ public String selfRegister(
         return PAGE_NOT_FOUND_VIEW;
     }
 
+    private void parseFormData(String formData, Model model) {
+        if (formData != null) {
+            try {
+                final RegisterFormData data = mapper
+                    .readerFor(RegisterFormData.class)
+                    .readValue(Base64.decode(formData.getBytes()));
```

Comment:
Is this safe? As in, would it be possible for a potential attack to inject malicious code?

Reply:
good point, I think we need to change how we render it in the jsp to disable any scripts that could potentially be injected

Reply:
Try injecting this string into this method and look at the console output: 
`eyJmaXJzdE5hbWUiOiJKb2huIiwibGFzdE5hbWUiOiJEb2UiLCJlbWFpbCI6ImpvaG4uZG9lQGVtYWlsLmNvbSJ9`

Reply:
https://www.nccgroup.trust/globalassets/our-research/us/whitepapers/2018/jackson_deserialization.pdf

Reply:
I have added owasp html encoding